### PR TITLE
docs: usage of List in routes

### LINKED ISSUE #7788: Document usage of List in routes
[The docs](https://www.playframework.com/documentation/2.6.x/JavaRouting) mention offhand that `List[String]` is a valid parameter type, but don't say how to use it. I was trying a bunch of stuff like `param=a,b,c,d` that wasn't working. I finally saw somewhere that suggested the usage might be `param=a&param=b&param=c&param=d` (note I haven't tried that syntax, so am not 100% certain it works)

### PR DESCRIPTION
**Pull request by amtj**
fixes #7788

### CODE REVIEW

File: documentation/manual/working/javaGuide/main/http/JavaRouting.md [hunk 1]
```diff
@@ -121,7 +121,20 @@ Then use the same type for the corresponding action method parameter in the cont
 
 @[clients-show-action](code/javaguide/http/routing/controllers/Clients.java)
 
-> **Note:** The parameter types are specified using a suffix syntax. Also, the generic types are specified using the `[]` symbols instead of `<>`, as in Java. For example, `List[String]` is the same type as the Java `List<String>`.
+The parameter types are specified using a suffix syntax. Also, the generic types are specified using the `[]` symbols instead of `<>`, as in Java. For example, `List[String]` is the same type as the Java `List<String>`. The syntax is `param[0]=a&param[1]=b&param[2]=c` to have orderered parameters or `param[]=a&param[]=b&param[]=c` which does not guarantee any parameter order.
```
Comment (gmethvin) [inline documentation/manual/working/javaGuide/main/http/JavaRouting.md]: Have you tested that this is correct? I don't believe you can do the `param[0]=a&param[1]=b` syntax. all the keys have to have the same name, e.g. `param=a&param=b`

Comment (amtj) [reply documentation/manual/working/javaGuide/main/http/JavaRouting.md]: @gmethvin I tried to test the syntax. But failed to get results even with `param=a&param=b` syntax.  I am using sbt 0.13.15 can it be a problem?

Also, I found some information [here](https://groups.google.com/forum/?fromgroups#!topic/play-framework/c5kB6wmcF8Q) supporting the argument but without any evidence.

@marcospereira Have you tested it?

Comment (marcospereira) [reply documentation/manual/working/javaGuide/main/http/JavaRouting.md]: Sorry for taking soo long to reach back to you, @amtj.

I'm reading the issue now, and there was some confusion. The syntax I've explained is for Form `POST` requests while the issue is talking about query strings in a `GET` request. So, for `GET` request the valid syntax is `param=a&param=b` while for `POST` you can do as I've explained at issue.

**Comment by marcospereira** [comment]
@amtj let me know if you are still interested in pushing this forward. If so, I will do my best to review it in a timely manner.

File: documentation/manual/working/javaGuide/main/http/JavaRouting.md (hunk 1 above)
Comment (mkurz) [reply documentation/manual/working/javaGuide/main/http/JavaRouting.md]: Also see https://github.com/playframework/playframework/issues/7788#issuecomment-540986731

**Comment by mkurz** [comment]
Fixed by #9716 already (also this pull request was wrong anyway)

**Pull request closed without merging**